### PR TITLE
Stop testing unmaintained versions

### PR DESCRIPTION
**Pull request by greg0ire**
Refs #285

### CODE REVIEW

File: config/projects.yml [hunk 1]
```diff
@@ -403,9 +405,9 @@ user-bundle:
         sonata_core: ['3']
         sonata_admin: ['3']
     3.x:
-      php: ['5.3', '5.4', '5.5', '5.6', '7.0', '7.1']
+      php: ['5.6', '7.0', '7.1']
       versions:
-        symfony: ['2.3', '2.7', '2.8']
+        symfony: ['2.8']
```
Comment (core23) [inline config/projects.yml]: Please add symfony 3.3, so we have at least one symfony 3 release remaining

File: config/projects.yml [hunk 2]
```diff
@@ -387,9 +389,9 @@ translation-bundle:
         sonata_core: ['3']
         sonata_admin: ['3']
     2.x:
-      php: ['5.4', '5.5', '5.6', '7.0', '7.1']
+      php: ['5.6', '7.0', '7.1']
       versions:
-        symfony: ['2.3', '2.7', '2.8', '3.2', '3.3']
+        symfony: ['2.8', '3.1', '3.2']
```
Comment (core23) [inline config/projects.yml]: Symfony 3.1 is EOL

File: config/projects.yml [hunk 3]
```diff
@@ -157,10 +157,10 @@ doctrine-mongodb-admin-bundle:
         symfony: ['2.8', '3.2', '3.3']
         sonata_admin: ['3']
     3.x:
-      php: ['5.3', '5.4', '5.5', '5.6']
+      php: ['5.6']
```
Comment (core23) [inline config/projects.yml]: What about PHP 7.1?

File: config/projects.yml (hunk 2 above)
Comment (greg0ire) [reply config/projects.yml]: Woops that one was unintended

Comment (greg0ire) [reply config/projects.yml]: Had an old version, and made my changes before rebasing.

File: config/projects.yml (hunk 1 above)
Comment (greg0ire) [reply config/projects.yml]: Is there a proof PR for that? I'm not sure the user bundle is supposed to work with sf3 in version 2

File: config/projects.yml (hunk 3 above)
Comment (greg0ire) [reply config/projects.yml]: Give me a proof PR and I will add it.

File: config/projects.yml (hunk 1 above)
Comment (core23) [reply config/projects.yml]: You're right. In that case 👍

**Comment by Soullivaneuh** [comment]
@greg0ire I think we can drop PHP 5.6 and 7.0 too as doctrine does.

The only LTS constraint described in #285 is about Symfony.

Except maybe if another sonata dependent project does not work on php 7.1?

**Comment by greg0ire** [comment]
On the stable branches? I think it's nice to keep at least the latest minor of php we started the branch with. I'm more ok with dropping 7.0 than 5.6... Although 7.0 still has active support, and 5.6 does not.

**Comment by Soullivaneuh** [comment]
> On the stable branches? I think it's nice to keep at least the latest minor of php we started the branch with. I'm more ok with dropping 7.0 than 5.6... Although 7.0 still has active support, and 5.6 does not.

As discussed on slack, we keep support only on the green section: http://php.net/supported-versions.php

For 7.0, we will exceptionally drop it as it's already done on some projects.

File: config/projects.yml [hunk 4]
```diff
@@ -22,7 +22,7 @@ admin-search-bundle:
         sonata_admin: ['3']
         ruflin_elastica: ['2']
     1.x:
-      php: ['5.6', '7.0']
+      php: ['7.0']
```
Comment (Soullivaneuh) [inline config/projects.yml]: 7.1 should be supported: https://github.com/sonata-project/SonataAdminSearchBundle/blob/9c3496449e4a6d21a8a196548b378bceb74fa9ad/composer.json#L20

**Pull request merged**